prefect_kubernetes/pods.py

```diff
@@ -0,0 +1,361 @@
+    kubernetes_credentials: KubernetesCredentials,
+    pod_name: str,
+    container: str,
+    namespace: Optional[str] = "default",
+    print_func: Optional[Callable] = None,
+    **kube_kwargs: Dict[str, Any],
+) -> Union[str, None]:
+    """Read logs from a Kubernetes pod in a given namespace.
+
+    Args:
+        kubernetes_credentials: `KubernetesCredentials` block for creating
+            authenticated Kubernetes API clients.
+        pod_name: The name of the pod to read logs from.
+        container: The name of the container to read logs from.
+        namespace: The Kubernetes namespace to read this pod from.
+        print_func: If provided, it will stream the pod logs by calling `print_func`
+            for every line and returning `None`. If not provided, the current pod
+            logs will be returned immediately.
+        **kube_kwargs: Optional extra keyword arguments to pass to the Kubernetes API.
+
+    Returns:
+        A string containing the logs from the pod's container.
+
+    Example:
+        Read logs from a pod in the default namespace:
+        ```python
+        from prefect import flow, get_run_logger
+        from prefect_kubernetes.credentials import KubernetesCredentials
+        from prefect_kubernetes.pods import read_namespaced_pod_logs
+
+        @flow
+        def kubernetes_orchestrator():
+            logger = get_run_logger()
+
+            pod_logs = read_namespaced_pod_logs(
+                kubernetes_credentials=KubernetesCredentials.load("k8s-creds"),
+                pod_name="test-pod",
+                container="test-container",
+                print_func=logger.info
+            )
+        ```
+    """
+    with kubernetes_credentials.get_client("core") as core_v1_client:
+
+        if print_func is None:
+            return await run_sync_in_worker_thread(
+                core_v1_client.read_namespaced_pod_log,
+                name=pod_name,
+                namespace=namespace,
+                container=container,
+                **kube_kwargs,
+            )
+
+        # From the `kubernetes.watch` documentation:
+        # Note that watching an API resource can expire. The method tries to
+        # resume automatically once from the last result, but if that last result
+        # is too old as well, an `ApiException` exception will be thrown with
+        # ``code`` 410.
+
+        while True:
+            try:
+                for log_line in Watch().stream(
+                    core_v1_client.read_namespaced_pod_log,
+                    name=pod_name,
+                    namespace=namespace,
+                    container=container,
+                ):
+                    print_func(log_line)
+                return
+
+            except ApiException as e:
+                if e.status != 410:
+                    raise
+
+
+@task
+async def replace_namespaced_pod(
+    kubernetes_credentials: KubernetesCredentials,
+    pod_name: str,
+    body: V1Pod,
```

Comment:
I think we can give a clearer name to this parameter. `new_pod` is what came to my mind, but there might be something better.
```suggestion
    new_pod: V1Pod,
```

Reply:
If we name the above `pod_config`, I vote for consistency.